aiohttp/connector.py

```diff
@@ -1174,6 +1178,10 @@ def drop_exception(fut: "asyncio.Future[List[Dict[str, Any]]]") -> None:
             host = hinfo["host"]
             port = hinfo["port"]
 
+            server_hostname = hinfo["hostname"] if sslcontext else None
+            if req.server_hostname is not None:
+                server_hostname = req.server_hostname
```

Comment:
Should this be set regardless of `sslcontext`? Seems like a change in behaviour there.

Reply:
> Should only be passed if ssl is not None.

So, looks like not.

Reply:
Implemented the one-liner suggestion below :+1: